src/models/Scheduler.php

```diff
@@ -145,21 +145,14 @@ public function readAll(?QueryParamsInterface $queryParams = null): array
             $this->appendFilterSql(column: 'team_events.userid', paramName: 'ownerid', value: $queryParams->getQuery()->getInt('eventOwner'));
             // handle multiple items
             $itemParams = $queryParams->getQuery()->all('items');
-            if (!empty($itemParams)) {
-                $ids = array_filter(array_map('intval', $itemParams)); // force numeric IDs
-                if ($ids) {
-                    $placeholders = array();
-                    foreach ($ids as $index => $id) {
-                        $key = "itemid$index";
-                        $placeholders[] = ":$key";
-                        $this->filterBindings[$key] = $id;
-                    }
-                    $this->filterSqlParts[] = 'AND items.id IN (' . implode(',', $placeholders) . ')';
-                }
-            }
+            $ids = array_filter(array_map('intval', $itemParams)); // force numeric IDs
+            if ($ids) {
```

Comment:
you can move the if into the function that processes it, so caller doesn't have to check emptyness, but the function using the value yes